discv5/discv5-theory.md

```diff
@@ -92,141 +92,199 @@ from the `k` closest nodes it has seen.
+    req-time     = absolute time of REGTOPIC request
+    wait-time    = waiting time assigned when ticket was created
+    cum-wait     = cumulative waiting time of this node
+
+### Registration Window
+
+The image below depicts a single ticket's validity over time. When the ticket is issued,
+the node keeping it must wait until the registration window opens. The length of the
+registration window is 10 seconds. The ticket becomes invalid after the registration
+window has passed.
+
+![ticket validity over time](./img/ticket-validity.png)
+
+Since all ticket waiting times are assigned to expire when a slot in the queue opens, the
+advertisement medium may receive multiple valid tickets during the registration window and
+must choose one of them to be admitted in the topic queue. The winning node is notified
+using a [REGCONFIRMATION] response.
+
+Picking the winner can be achieved by keeping track of a single 'next ticket' per queue
+during the registration window. Whenever a new ticket is submitted, first determine its
+validity and compare it against the current 'next ticket' to determine which of the two is
+better according to an implementation-defined metric such as the cumulative wait time
+stored in the ticket.
 
 ### Advertisement Protocol
 
-Let us assume that node `A` advertises itself under topic `T`. It selects node `C` as
-advertisement medium and wants to register an ad, so that when node `B` (who is searching
-for topic `T`) asks `C`, `C` can return the registration entry of `A` to `B`.
+This section explains how the topic-related protocol messages are used to place an ad.
 
-Node `A` first tells `C` that it wishes to register by requesting a ticket for topic `T`,
-using the [REQTICKET] message.
+Let us assume that node `A` provides topic `T`. It selects node `C` as advertisement
+medium and wants to register an ad, so that when node `B` (who is searching for topic `T`)
+asks `C`, `C` can return the registration entry of `A` to `B`.
 
-    A -> C  REQTICKET
+Node `A` first attempts to register without a ticket by sending [REGTOPIC] to `C`.
 
-`C` replies with a ticket. The ticket contains the node identifier of `A`, the topic, a
-serial number and wait period assigned by `C`.
+    A -> C  REGTOPIC [T, ""]
 
-    A <- C  TICKET
+`C` replies with a ticket and waiting time.
 
-Node `A` now waits for the duration of the wait period. When the wait is over, `A` sends a
-registration request including the ticket. `C` does not need to remember its issued
-tickets, just the serial number of the latest ticket accepted from `A` (after which it
-will not accept any tickets issued earlier).
+    A <- C  TICKET [ticket, wait-time]
 
-    A -> C  REGTOPIC
+Node `A` now waits for the duration of the waiting time. When the wait is over, `A` sends
+another registration request including the ticket. `C` does not need to remember its
+issued tickets since the ticket is authenticated and contains enough information for `C`
```

Comment:
Here you claim that `C` does not need to remember tickets, which then assumes that there is no server side 'session'

Reply:
Yeah, maybe the HTTP session analogy isn't the best. I'll replace it with something else.

Reply:
Tickets are more like JWTs